src/main/java/io/appform/dropwizard/actors/connectivity/RMQConnection.java

```diff
@@ -160,25 +161,56 @@ public void addBinding(String queueName, String exchange, String routingKey) thr
     public Map<String, Object> rmqOpts(final ActorConfig actorConfig) {
         final Map<String, Object> ttlOpts = getActorTTLOpts(actorConfig.getTtlConfig());
         final Map<String, Object> priorityOpts = getPriorityOpts(actorConfig);
-        return ImmutableMap.<String, Object>builder()
+        Builder<String, Object> builder = ImmutableMap.<String, Object>builder()
                 .putAll(ttlOpts)
-                .putAll(priorityOpts)
-                .put("x-ha-policy", "all")
-                .put("ha-mode", "all")
-                .build();
+                .putAll(priorityOpts);
+        switch (actorConfig.getQueueType()) {
```

Comment:
switch without a return statement does not break compilation when a new enum value is added. Maybe use classic enum visitor pattern here

Reply:
Makes sense, have added the visitor pattern. Code looks much cleaner now.